binfmt/binfmt_execmodule.c

```diff
@@ -400,17 +398,6 @@ int exec_module(FAR struct binary_s *binp,
     }
 #endif
 
-  /* Perform file actions */
-
-  if (actions != NULL)
```

Comment:
You cannot move the actions-pointer to nxtask_init call, as the pointer points to the wrong address environment.

On line 270
ret = addrenv_select(binp->addrenv, &binp->oldenv);

And its counterpart on line 395: 
ret = addrenv_restore(binp->oldenv);

You either have to make a kernel addressable copy of the actions pointer, or leave the file action handling here (after addrenv restore).

Reply:
emm... you are right, I forgot to consider kernel mode compatibility.

Reply:
@pussuw As your suggestion, I made a addressable copy of the action list in kernel mode, please review again.